plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -477,6 +488,19 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
             } );
 
       // relay signals to channels
+      my->pre_accepted_block_connection = my->chain->pre_accepted_block.connect([this](const signed_block_ptr& blk) {
+         auto itr = my->loaded_checkpoints.find( blk->block_num() );
+         if( itr != my->loaded_checkpoints.end() ) {
+            auto id = blk->id();
+            EOS_ASSERT( itr->second == id, checkpoint_exception,
```

Comment:
In the future it would probably be convenient to allow for short-ids however, since we encode the block number in the ID we would need a canonical short ID that is reasonably collision resistant first.